File: playwright_python/redbuspro.py
from playwright.sync_api import Playwright, sync_playwright
from playwright_stealth import stealth_sync  # use the sync version

def run(playwright: Playwright) -> None:
    browser = playwright.chromium.launch(headless=False)
    context = browser.new_context(locale="en-US")  # valid locale
    page = context.new_page()

    stealth_sync(page)  # stealth applied here

    page.goto("https://accounts.redbus.com/login?continue=https://www.redbus.pro/")
    page.get_by_role("textbox", name="Username").click()
    page.get_by_role("textbox", name="Username").fill("9748")
    page.get_by_role("textbox", name="Username").press("Tab")
    page.get_by_placeholder("Password").fill("prasannapurple")
    page.get_by_role("button", name="Login", exact=True).click()

    page.wait_for_timeout(10000)  # wait for UI to load or debug visually

    context.close()
    browser.close()

with sync_playwright() as playwright:
    run(playwright)

File: playWright/redbuspro.js
const { chromium } = require('playwright');
const { convert } = require('html-to-text');
const stealth = require('playwright-extra-plugin-stealth')();

chromium.use(stealth);


async function searchBuses(source, destination, operator = null) {
  const browser = await chromium.launch({ headless: false });
  const context = await browser.newContext({
    userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 Chrome/122.0.0.0 Safari/537.36',
    viewport: { width: 1280, height: 800 }
  });
  
  const page = await context.newPage();

  await page.goto('https://accounts.redbus.com/login?continue=https://www.redbus.pro/');
  await page.getByRole('textbox', { name: 'Username' }).click();
  await page.getByRole('textbox', { name: 'Username' }).fill('9748');
  await page.getByRole('textbox', { name: 'Username' }).press('Tab');
  await page.getByRole('textbox', { name: 'Password' }).click();
  await page.getByRole('textbox', { name: 'Password' }).fill('prasannapurple');
  await page.getByRole('button', { name: 'Login', exact: true }).click();


  await browser.close();
}


const routes = [
    { source: 'Pune', destination: 'Goa', operator: 'Prasanna - Purple Bus' },
    { source: 'Mumbai', destination: 'Goa' }
  ];
  
  (async () => {
    for (const route of routes) {
      await searchBuses(route.source, route.destination, route.operator);
    }
  })();
  